otcextensions/sdk/cce/v3/node_pool.py

```diff
@@ -0,0 +1,223 @@
+    # Description about the primary Nic
+    primary_nic = resource.Body('primaryNic', type=SubnetIdSpec)
+
+
+class NodeTemplateSpec(resource.Resource):
+    # Properties
+    # Name of the AZ where the node resides.
+    # If availability_zone is set to: random, nodes can be created in any AZ
+    availability_zone = resource.Body('az')
+    # Billing mode of a node. Currently, only pay-per-use is supported.
+    billing_mode = resource.Body('billingMode', type=int, default=0)
+    # Number of nodes.
+    count = resource.Body('count', type=int)
+    # Data disk parameters of a node. At present, only one data
+    # disk can be configured
+    data_volumes = resource.Body('dataVolumes', type=list,
+                                 list_type=DataVolumeSpec)
+    # ID of the Dedicated Host to which nodes will be scheduled
+    dedicated_host = resource.Body('dedicatedHostId')
+    # ID of the ECS group where the CCE node can belong to
+    ecs_group = resource.Body('ecsGroupId')
+    # Extended parameters in key-value format
+    extend_params = resource.Body('extendParam', type=ExtendParamSpec)
+    # The node is created in the specified fault domain.
+    fault_domain = resource.Body('faultDomain')
+    # Flavor (mandatory)
+    flavor = resource.Body('flavor')
+    # Elastic IP address parameters of a node.
+    floating_ip = resource.Body('publicIP', type=PublicIPSpec)
+    # Kubernetes tags
+    k8s_tags = resource.Body('k8sTags', type=dict)
+    # Parameters for logging in to the node.
+    login = resource.Body('login')
+    node_nic_spec = resource.Body('nodeNicSpec', type=NodeNicSpec)
+    # Boolean: if node is offloading all its components
+    offload_node = resource.Body('offloadNode', type=bool)
+    # Operating System of the node. EulerOS and CentOS are supported.
+    os = resource.Body('os')
+    # System disk parameters of the node.
+    root_volume = resource.Body('rootVolume', type=VolumeSpec)
+    # Tags of a Node
+    tags = resource.Body('userTags', type=list, list_type=TagSpec)
+    # Taints are used to configure anti-affinity
+    taints = resource.Body('taints', type=list, list_type=TaintSpec)
+
+
+class NodePoolSpec(resource.Resource):
+    # Autoscaling parameters
+    autoscaling = resource.Body('autoscaling', type=AutoScalingSpec)
+    # Expected number of nodes in this node pool.
+    initial_node_count = resource.Body('initialNodeCount', type=int)
+    node_management = resource.Body('nodeManagement', type=NodeManagementSpec)
+    # Node pool type. Currently only ECSs are supported
+    # Value: vm
+    node_pool_type = resource.Body('type')
+    # Template of the node specification.
+    node_template_spec = resource.Body('nodeTemplate', type=NodeTemplateSpec)
+
+
+class MetaDataSpec(resource.Body):
+    # Name of the node pool
+    name = resource.Body('name')
+    #: UUID
+    #: *Type:str
+    id = resource.Body('uid', alternate_id=True)
+
+
+class NodePool(_base.Resource):
+
```

Comment:
resource_key/keys? not needed here?

Reply:
no, there is no resource key given by IP request / response. Same with cluster node.